internal/cli/atlas/onlinearchive/delete.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package onlinearchive
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/mongodb/mongocli/internal/validate"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.GlobalOpts
+	*cli.DeleteOpts
+	clusterName string
+	store       store.OnlineArchiveDeleter
+}
+
+func (opts *DeleteOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteOnlineArchive, opts.ConfigProjectID(), opts.clusterName)
+}
+
+// mongocli atlas cluster(s) onlineArchive(s) delete <id> [--clusterName name][--projectId projectId][--force]
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts("Archive '%s' deleted\n", "Archive not deleted"),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <id>",
```

Comment:
[q] Are clusterName and projectId not always required? This seems to read as those are optional parameters, but that's not the case here: https://github.com/mongodb/go-client-mongodb-atlas/blob/master/mongodbatlas/online_archives.go#L137-L144

Reply:
that's actually a fair point, I've been forgetting to mark the flag as required and the copy/past error has continued 
good catch

Reply:
@colm-quinn fixed here and other commands

Reply:
Cool, looks good.
There's still a few small questions from me on this, part is I'm probably missing what cobra is doing behind the scenes.

1. Is the required on the `projectID` inferred from another place, as I don't see it here?
2. I'm still unsure about what gets printed to screen here. The method comment has `delete <id> [--clusterName name][--projectId projectId][--force]` but the `Use:     "delete <id>"` above doesn't imply to me that they will show in the help.

I'm assuming there's some magic in cobra that does this, but would be good to confirm.

Reply:
1. Project ID is more complex because it can come from a flag, env variable or your profile file, so what I've done is a little shared function here https://github.com/mongodb/mongocli/blob/55cce5ce4dbf25efa84fc3301f5e42ead7a3261b/internal/cli/global_opts.go#L53, that validates is present via any of the available options, and is a valid `ObjectId`, we use this function here on line 56

2. On the `Use:` property we only define things that are passed as args and not flags, cobra doesn't enforce this and we do it as a nice to have as when you print usage you get
```
mongocli atlas clusters onlineArchives delete <id> [flags]
```
We don't define flags there, cobra already adds a ` [flags]` as you can see, and since every flag has help you get their help detailed, heres the full example of this command
```
mongocli atlas clusters onlinearchive rm --help
Delete an online archives from a cluster.

Usage:
  mongocli atlas clusters onlineArchives delete <id> [flags]

Aliases:
  delete, rm

Flags:
      --clusterName string   Name of the cluster.
      --force                Don't ask for confirmation.
  -h, --help                 help for delete
      --projectId string     Project ID to use. Overrides configuration file or environment variable settings.

Global Flags:
  -P, --profile string   Profile to use from your configuration file.
```

Reply:
doing this helped me spot a typo on the description of the command

Reply:
Excellent, thanks for the detailed explanation - that makes it a lot clearer.